input: misc: qcom-spmi-haptics: make auto res mode and vmax configureable via DT prop

Smankusors opened the pull request:
This has been tested on my Zenfone 3 ZE520KL with pmi8952, and it works as I wanted using `qcom,auto-res-mode = <HAP_AUTO_RES_QWD>` and `qcom,vmax-millivolt = <1500>`.

On another note, I noticed that the pm660 has only two available auto res modes to select:  
https://github.com/Zenfone-5-X00QD-4-19/kernel_asus_sdm660/blob/3163b5f4e036f69e43626ae60f4b67b8bb61b733/drivers/soc/qcom/qpnp-haptic.c#L232-L235  

I think supporting the pm660 is outside the scope of this PR, and the haptics on my Zenfone 5 with pm660 seem to be functioning normally anyway.

Wdyt?

barni2000 commented:
I think one of your commit prefix is not correct. Check this https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/Documentation/devicetree/bindings/submitting-patches.rst

Smankusors commented:
> I think one of your commit prefix is not correct. Check this https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/Documentation/devicetree/bindings/submitting-patches.rst
> 
> You should separte schema modifications from the code modification

ok, updated and rebased 👍

barni2000 commented:
Thanks!
